pysph/sph/wc/crksph.py

```diff
@@ -741,9 +872,47 @@ def get_particle_array_crksph(constants=None, **props):
     return pa
 
 
+class CRKSPHIntegrator(Integrator):
+    def one_timestep(self, t, dt):
+        self.stage1()
+        self.do_post_stage(dt, 1)
+
+        self.compute_accelerations(0)
+
+        self.stage2()
+        # We update domain here alone as positions only change here.
+        # self.update_domain()
```

Comment:
Why is this commented?

Reply:
because we only update velocity in stage2() for the energy equation, after the update is done, we use stage3() to update the position and energy, will clean this up